jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -1100,29 +1220,55 @@ public MethodHandle findStaticSetter(Class<?> clazz, String fieldName, Class<?>
 			return handle;
 		}
 		
+		/*[IF Java14]*/
 		/**
 		 * Create a lookup on the request class.  The resulting lookup will have no more 
 		 * access privileges than the original.
 		 * 
 		 * @param lookupClass - the class to create the lookup on
 		 * @return a new MethodHandles.Lookup object
+		 * @throws NullPointerException - if lookupClass is null
+		 * @throws IllegalArgumentException - if the requested Class is a primitive type or an array class
 		 */
-		public MethodHandles.Lookup in(Class<?> lookupClass){
-			lookupClass.getClass();	// implicit null check
+		public MethodHandles.Lookup in(Class<?> lookupClass) throws NullPointerException, IllegalArgumentException {
+		/*[ELSE]*/
+		/**
+		 * Create a lookup on the request class.  The resulting lookup will have no more 
+		 * access privileges than the original.
+		 * 
+		 * @param lookupClass - the class to create the lookup on
+		 * @return a new MethodHandles.Lookup object
+		 */
+		public MethodHandles.Lookup in(Class<?> lookupClass) {
+		/*[ENDIF] Java14 */
+			if (Objects.isNull(lookupClass)) {
```

Comment:
Use `Objects.requireNonNull(lookupClass)` here rather than the explicit method call for a null check

Reply:
Agreed and fixed.